simde/x86/sse2.h

```diff
@@ -7290,7 +7290,7 @@ simde_mm_unpacklo_pd (simde__m128d a, simde__m128d b) {
 SIMDE_FUNCTION_ATTRIBUTES
 simde__m128d
 simde_x_mm_negate_pd(simde__m128d a) {
-  #if defined(SIMDE_X86_SSE_NATIVE)
+  #if defined(SIMDE_X86_SSE2_NATIVE)
```

Comment:
huh, I wonder why our testing didn't catch this..

Reply:
I also get some very surprising errors in several NEON functions involving f64 if I disable SSE2.
I am still trying to understand why they occur :-(

Reply:
Actually it doesn't seem to be as simple as that.
In this case where the failing addition gives -NAN, if the same addition is repeated in the loop it seems to give the correct result. Could it be a race condition between data being loaded into a_ and b_ (or r_) and their use ?
Maybe a tiny sleep would make the problem disappear.
Any better ideas?

Reply:
A race condition wouldn't make sense; the code isn't multi-threaded (the tests are run in parallel, but the parallelism is process-level not thread-level).

It sounds like a compiler bug, which I wouldn't really find surprising here.  I'm guessing you're targeting x86_64, and the baseline there includes SSE2 so I have a feeling -mno-sse2 isn't particularly well-tested.  If I can reproduce I'll try to set c-reduce loose on it and see what it comes up with.

Reply:
The sleep didn't help.
However if I simply duplicate the  `r_.values[i] = a_.values[i] + b_.values[i];` line:
- The error goes away in C
- I get a different equally strange error in C++:
`test/arm/neon/add.cpp:638: assertion failed: r[0] ~= simde_vld1q_f64(test_vec[i].r)[0] (99.060000 ~= 99.060000)`

Incidentally I force the portable code with -DSIMDE_NO_NATIVE but the initial error is also present without it.

Reply:
> Concerning the race condition: I imagined that something might cause the hardware to ignore the latency of loading from memory. IIRC some asynchronous load and cache pre-load instructions exist.

Thanks for explaining, that makes more sense though it would still be a compiler bug since it violates sequential consistency.

> However if I simply duplicate the r_.values[i] = a_.values[i] + b_.values[i]; line:

Good data point, thanks.  If manually unrolling the loop fixes the issue in C I'm even more convinced of a compiler bug.

> I get a different, but equally strange, error in C++:

Yep, I've seen that before. Compiler bug.  I have two others to try to generate a reduced test case for, I'll try to take a look at this one once those are done.  Thanks for letting me know.

Reply:
This is the list of functions with a similar problem, all in the *q_f64 intrinsic:
`arm/neon/{add,bsl,ceqz,cgez,cle,clez,clt,cltz,min,mls,mul,mul_n,padd,sub}`

Reply:
My impression is that sometimes, but consistently, some values are read incorrectly the first time but a subsequent read gives the correct value. This would also explain why two apparently equal values are declared unequal in the assertion.